cf-java-logging-support-servlet-jakarta/pom.xml

```diff
@@ -0,0 +1,194 @@
+			<groupId>com.sap.hcp.cf.logging</groupId>
+			<artifactId>cf-java-logging-support-servlet</artifactId>
+			<version>${project.version}</version>
+			<classifier>tests</classifier>
+			<scope>provided</scope>
+		</dependency>
+
+		<!-- servlet api -->
+		<dependency>
+			<groupId>jakarta.servlet</groupId>
+			<artifactId>jakarta.servlet-api</artifactId>
+			<version>${servlet.api.version}</version>
+			<scope>provided</scope>
+		</dependency>
+		<dependency>
+			<groupId>com.sap.hcp.cf.logging</groupId>
+			<artifactId>cf-java-logging-support-core</artifactId>
+			<version>${project.version}</version>
+		</dependency>
+		<!-- we need our logback implementation for testing! -->
+		<dependency>
+			<groupId>com.sap.hcp.cf.logging</groupId>
+			<artifactId>cf-java-logging-support-logback</artifactId>
+			<version>${project.version}</version>
+			<scope>test</scope>
+		</dependency>
+		<dependency>
+			<groupId>ch.qos.logback</groupId>
+			<artifactId>logback-classic</artifactId>
+			<version>${logback.version}</version>
+			<scope>test</scope>
+		</dependency>
+		<!-- Library for token signing/verification -->
+		<dependency>
+			<groupId>com.auth0</groupId>
+			<artifactId>java-jwt</artifactId>
+			<version>3.18.2</version>
+		</dependency>
+		<dependency>
+			<groupId>com.fasterxml.jackson.core</groupId>
+			<artifactId>jackson-databind</artifactId>
+			<version>2.13.4.1</version>
+		</dependency>
+
+		<!-- testing -->
+		<dependency>
+			<groupId>org.eclipse.jetty</groupId>
+			<artifactId>jetty-server</artifactId>
+			<version>11.0.13</version>
+			<scope>test</scope>
+		</dependency>
+		<dependency>
+			<groupId>org.eclipse.jetty</groupId>
+			<artifactId>jetty-servlet</artifactId>
+			<version>11.0.13</version>
+			<scope>test</scope>
+		</dependency>
+		<dependency>
+			<groupId>org.apache.httpcomponents</groupId>
+			<artifactId>httpclient</artifactId>
+			<version>4.5.13</version>
+			<scope>test</scope>
+		</dependency>
+
+	</dependencies>
+
+	<build>
+		<plugins>
+			<plugin>
+				<artifactId>maven-clean-plugin</artifactId>
+				<version>3.2.0</version>
+				<configuration>
+					<filesets>
+						<fileset>
+							<directory>${basedir}/src</directory>
+							<followSymlinks>false</followSymlinks>
+						</fileset>
+					</filesets>
+				</configuration>
+			</plugin>
+
+			<plugin>
+				<artifactId>maven-dependency-plugin</artifactId>
+				<version>3.5.0</version>
+				<executions>
+					<execution>
+						<id>unpack</id>
+						<phase>generate-sources</phase>
+						<goals>
+							<goal>unpack</goal>
+						</goals>
+						<configuration>
+							<artifactItems>
+								<artifactItem>
+									<groupId>${project.groupId}</groupId>
+									<artifactId>cf-java-logging-support-servlet</artifactId>
+									<version>${project.version}</version>
+									<type>jar</type>
+									<classifier>sources</classifier>
+									<overWrite>false</overWrite>
+									<outputDirectory>${basedir}/src/main/java</outputDirectory>
```

Comment:
Why did put the sources in the main source directory? Since these sources are generated, they could have gone to `target/generated-sources/` instead.

Reply:
Is this folder part of the classpath? The folders src/main/java and src/test/java are out-of-box part of the classpath and we don't have to configure additional source and test-source paths. This can be done, but makes the pom.xml more complex.

Reply:
My proposal is to use target/generated-(test-)sources and target generated-(test-)-resources to NOT check in generated code into github.
The out-of-the-bos behaviour then can get achieved by this:
```
        <plugin>
          <groupId>org.codehaus.mojo</groupId>
          <artifactId>build-helper-maven-plugin</artifactId>
          <version>${build-helper-maven-plugin.version}</version>
          <executions>
            <execution>
              <id>add-generated-sources</id>
              <goals>
                <goal>add-source</goal>
              </goals>
              <phase>generate-sources</phase>
              <configuration>
                <sources>
                  <source>target/generated-sources/java</source>
                </sources>
              </configuration>
            </execution>
            <execution>
              <id>add-generated-resources</id>
              <goals>
                <goal>add-resource</goal>
              </goals>
              <phase>generate-resources</phase>
              <configuration>
                <resources>
                  <resource>
                    <directory>target/generated-sources/resources</directory>
                  </resource>
                </resources>
              </configuration>
            </execution>
            <execution>
              <id>add-generated-test-sources</id>
              <goals>
                <goal>add-test-source</goal>
              </goals>
              <phase>generate-test-sources</phase>
              <configuration>
                <sources>
                  <source>target/generated-test-sources/java</source>
                </sources>
              </configuration>
            </execution>
            <execution>
              <id>add-generated-test-resources</id>
              <goals>
                <goal>add-test-resource</goal>
              </goals>
              <phase>generate-test-resources</phase>
              <configuration>
                <resources>
                  <resource>
                    <directory>target/generated-test-sources/resources</directory>
                  </resource>
                </resources>
              </configuration>
            </execution>
          </executions>
        </plugin>
```

Reply:
I hat put the src folder to gitignore, so it was not pushed to github. But I can use build-helper-maven-plugin  if it's preferred.

Reply:
both ways work, was just a proposal, let's have @KarstenSchnitter decide what to use :-)

Reply:
I was alluding to the solution proposed by @aamotharald. I prefer having the code in the generated-sources folders. That way clearly communicates, that the code should not be changed in an IDE.

Reply:
ok, reworked as requested.